### PR TITLE
Better support for Cucumber framework

### PR DESCRIPTION
**Pull request by mikesalvia**
## Proposed changes

Add better codeRef for Cucumber-style testing
Mark scenario SKIPPED if all children are SKIPPED

### CODE REVIEW

**Comment by mikesalvia** [comment]
@BorisOsipov I will work on fixing unit tests.

File: lib/reporter.ts [hunk 1]
```diff
@@ -258,6 +262,9 @@ class ReportPortalReporter extends Reporter {
     this.sessionId = runner.sessionId;
     this.client = client || new ReportPortalClient(this.options.reportPortalClientConfig);
     this.launchId = process.env.RP_LAUNCH_ID;
+    if (this.options.writeLaunchIdToFile) {
+      writeFileSync('./RP_LAUNCH_ID', this.launchId);
```
Comment (BorisOsipov) [inline lib/reporter.ts:266]: It should be done in service not in reporter https://github.com/BorisOsipov/wdio-reportportal-service/blob/master/src/index.js#L7

Comment (BorisOsipov) [reply lib/reporter.ts:266]: as well it should be service option

**Review by BorisOsipov — CHANGES_REQUESTED** [review]

File: lib/reporter.ts (hunk 1 above)
Comment (mikesalvia) [reply lib/reporter.ts:268]: I will migrate that over there.

**Comment by BorisOsipov** [comment]
@mikesalvia btw why do you need `launchId` saved to file? Just to make sure we will cover all such use cases.

**Comment by mikesalvia** [comment]
It is going to be used by a separate Bash command to add both re-run data as well as dry-runs for manual tests that will be reported on later.

File: lib/reporter.ts (hunk 1 above)
Comment (mikesalvia) [reply lib/reporter.ts]: Fixed

**Comment by mikesalvia** [comment]
I either plan to add a new plugin or make it a configurable option in [jwplayer/wdio-rerun-service](https://github.com/jwplayer/wdio-rerun-service).

**Comment by BorisOsipov** [comment]
@mikesalvia thank you, I get it. I will review\test this change in the next few days. Sorry for delay, I am a bit sick now(

**Comment by mikesalvia** [comment]
It is all good I am unblocked as I am using a custom build. 👍

**Comment by mikesalvia** [comment]
@BorisOsipov Checking back in on this one... Any other edits requested?

**Comment by BorisOsipov** [comment]
@mikesalvia I didn't have chance to go deeper into changes and review it. Hope I will have time in next two days.

I have only one question\request

Could you please have look into https://github.com/BorisOsipov/wdio-reportportal-reporter/pull/145 - it will be nice to make sure that your manipulation with coderef will always lead to the same codeRef for specific spec\scenario to not break RP history table.

**Comment by BorisOsipov** [comment]
merged in #146

**Comment by BorisOsipov** [comment]
@mikesalvia I a bit polish this changes to not break mocha\jasmine reporting.

Released in 6.4.0

**Pull request closed without merging**